src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/CaseFlag.java

```diff
@@ -1,33 +1,14 @@
 package uk.gov.hmcts.reform.iacaseapi.domain.entities;
 
-import static java.util.Objects.requireNonNull;
+import lombok.NonNull;
+import lombok.Value;
 
-import lombok.EqualsAndHashCode;
-import lombok.ToString;
-
-@EqualsAndHashCode
-@ToString
+@Value
 public class CaseFlag {
 
-    private CaseFlagType caseFlagType;
-    private String caseFlagAdditionalInformation;
-
-    private CaseFlag() {
-
-    }
-
-    public CaseFlag(CaseFlagType caseFlagType, String caseFlagAdditionalInformation) {
-        requireNonNull(caseFlagType);
-        requireNonNull(caseFlagAdditionalInformation);
-        this.caseFlagType = caseFlagType;
-        this.caseFlagAdditionalInformation = caseFlagAdditionalInformation;
-    }
-
-    public CaseFlagType getCaseFlagType() {
-        return caseFlagType;
-    }
+    @NonNull
+    CaseFlagType caseFlagType;
```

Comment:
Do we use CaseFlag in Prod already?
If yes, did you or Georgina test already existing flags?

Jackson parsing may brake already existing cases, when you mark it as @NonNull